src/main/java/teammates/ui/controller/InstructorCourseInstructorEditSaveAction.java

```diff
@@ -29,10 +28,29 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         gateKeeper.verifyAccessible(instructor, logic.getCourse(courseId),
                                     Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR);
 
-        InstructorAttributes instructorToEdit =
-                extractUpdatedInstructor(courseId, instructorId, instructorName, instructorEmail);
-        updateToEnsureValidityOfInstructorsForTheCourse(courseId, instructorToEdit);
+        String instructorId = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_ID);
+        InstructorAttributes instructorToEdit = instructorId == null
+                ? logic.getInstructorForEmail(courseId, instructorEmail)
+                : logic.getInstructorForGoogleId(courseId, instructorId);
+
+        boolean isDisplayedToStudentsAfterChanges =
+                getRequestParamValue(Const.ParamsNames.INSTRUCTOR_IS_DISPLAYED_TO_STUDENT) != null;
+        boolean isMakingInstructorInvisible = !isDisplayedToStudentsAfterChanges
+                && instructorToEdit.isDisplayedToStudents;
```

Comment:
More intuitive to read if you swap the conditions? i.e., `instructorToEdit.isDisplayedToStudents &&  !isDisplayedToStudentsAfterChanges`

Reply:
updated